core/query.go

```diff
@@ -27,8 +28,10 @@ func QueryClientStatePair(
 		if err != nil {
 			return err
 		}
-		path := host.FullClientStatePath(src.Path().ClientID)
-		srcCsRes.Proof, srcCsRes.ProofHeight, err = src.ProveState(srcCtx, path, value)
+		if prove {
+			path := host.FullClientStatePath(src.Path().ClientID)
+			srcCsRes.Proof, srcCsRes.ProofHeight, err = src.ProveState(srcCtx, path, value)
+		}
```

Comment:
You should also move marshaling of `srcCsRes.ClientState` into this if-prove statement.

Reply:
right

Reply:
@siburu fixed at https://github.com/hyperledger-labs/yui-relayer/pull/109/commits/d1b2beba50276c530041e60bac09ce3cff6f8c7b